src/components/forms/FormTimestampField.js

```diff
@@ -15,18 +15,26 @@ import TextField from "@material-ui/core/TextField";
 
 const onDateChange = (prevDate, fieldName, setFieldValue) => (event) => {
     const newValue = event.target.value;
-    const date = prevDate ? prevDate : new Date();
-    const time = formatDateObject(date, dateConstants.TIME_FORMAT);
+    if (newValue) {
+        const date = prevDate ? prevDate : new Date();
+        const time = formatDateObject(date, dateConstants.TIME_FORMAT);
 
-    setFieldValue(fieldName, `${newValue} ${time}`);
+        setFieldValue(fieldName, `${newValue} ${time}`);
+    } else {
+        setFieldValue(fieldName, "");
+    }
 };
 
 const onTimeChange = (prevDate, fieldName, setFieldValue) => (event) => {
     const newValue = event.target.value;
-    const date = prevDate ? prevDate : new Date();
-    const dateStr = formatDateObject(date, dateConstants.DATE_FORMAT);
+    if (newValue) {
+        const date = prevDate ? prevDate : new Date();
+        const dateStr = formatDateObject(date, dateConstants.DATE_FORMAT);
 
-    setFieldValue(fieldName, `${dateStr} ${newValue}`);
+        setFieldValue(fieldName, `${dateStr} ${newValue}`);
+    } else {
+        setFieldValue(fieldName, "");
+    }
```

Comment:
These changes made sense to me, but I wanted to point it out since we've had this component for a while and it hasn't been an issue. 
The issue I was facing was basically that if you filled in a value, changed your mind and cleared the value, part of the value would still remain and would be present in the `values` from Formik.